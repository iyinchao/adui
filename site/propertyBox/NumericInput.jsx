import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Select, Switch, Icon, Button, NumericInput } from "../../components"
import { SIZES, INTENTS } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    disabled: false,
    intent: "normal",
    size: "small",
    min: 0,
    max: 100,
  }

  render() {
    const { disabled, intent, size, min, max } = this.state

    const codes = `<NumericInput
  ${disabled ? "disabled" : ""}
  ${intent === "normal" ? "" : `intent="${intent}"`}
  ${size === "small" ? "" : `size="${size}"`}
  defaultValue={27}
  min={${min}}
  max={${max}}
/>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <NumericInput
              autoFocus
              style={{ width: "240px" }}
              disabled={disabled}
              intent={intent}
              size={size}
              defaultValue={27}
              min={min === "none" ? null : Number(min)}
              max={max === "none" ? null : Number(max)}
            />
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <dt className={styles.controlTitle}>类型</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {INTENTS.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === intent}
                      onClick={() => this.setState({ intent: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <dt className={styles.controlTitle}>最小值</dt>
              <dd className={styles.controlContent}>
                <Select
                  value={min}
                  onChange={(param) => this.setState({ min: param })}
                >
                  <Select.Option value="none">无</Select.Option>
                  <Select.Option value="-10">-10</Select.Option>
                  <Select.Option value="0">0</Select.Option>
                  <Select.Option value="20">20</Select.Option>
                </Select>
              </dd>
              <dt className={styles.controlTitle}>最大值</dt>
              <dd className={styles.controlContent}>
                <Select
                  value={max}
                  onChange={(param) => this.setState({ max: param })}
                >
                  <Select.Option value="none">无</Select.Option>
                  <Select.Option value="20">20</Select.Option>
                  <Select.Option value="50">50</Select.Option>
                  <Select.Option value="100">100</Select.Option>
                </Select>
              </dd>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="禁用态"
                  unCheckedText="禁用态"
                  checked={disabled}
                  onChange={(param) => this.setState({ disabled: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
