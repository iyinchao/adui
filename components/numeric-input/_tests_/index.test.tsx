/* eslint-disable */
import * as React from "react"
import TestRenderer from "react-test-renderer"
import NumericInput from ".."

// class Mounter extends React.Component<any, any> {
//   constructor(props) {
//     super(props)

//     this.state = {
//       value: 14,
//     }
//   }

//   render() {
//     const { value } = this.state
//     return (
//       <NumericInput
//         autoFocus
//         style={{ width: "100px" }}
//         value={value}
//         step={.01}
//         max={100}
//         min={0}
//         onChange={value => this.setState({ value })}
//       />
//     )
//   }
// }

describe("NumericInput", () => {
  beforeAll(() => {
    /**
     * https://stackoverflow.com/a/40006663
     * 通过 mock `getBoundingClientRect`，之后运行的所有 `getBoundingClientRect` 都会返回这同一个值。
     */
    Element.prototype.getBoundingClientRect = jest.fn(() => {
      return {
        width: 120,
        height: 120,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        x: 0,
        y: 0,
        toJSON: () => {},
      }
    })
  })

  it("快照验证", () => {
    const wrapper = TestRenderer.create(<NumericInput />)
    expect(wrapper).toMatchSnapshot()
  })
})
