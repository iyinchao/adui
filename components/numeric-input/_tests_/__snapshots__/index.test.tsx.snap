// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput 快照验证 1`] = `
<div
  className="adui-numeric-input adui-input-wrapper adui-input-small adui-input-normal"
  style={Object {}}
>
  <input
    autoComplete="off"
    autoFocus={false}
    className="adui-input-base"
    disabled={false}
    id={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="请输入"
    required={false}
    style={
      Object {
        "paddingLeft": undefined,
        "paddingRight": undefined,
      }
    }
    type={null}
    value=""
  />
  <div
    className="adui-input-right"
    onClick={[Function]}
    role="none"
  >
    <div
      className="adui-numeric-input-buttons"
    >
      <div
        className="adui-numeric-input-button"
        onMouseDown={[Function]}
        role="none"
      >
        <svg
          className="adui-icon-base adui-icon-interactive adui-icon-interactive_level_2"
          data-icon="arrow-up"
          data-interactive={true}
          fill="var(--gray-700)"
          height={12}
          onClick={[Function]}
          viewBox="0 0 18 18"
          width={12}
        >
          <path
            d="M14.303 11.3033L13.9495 11.6569C13.7542 11.8521 13.4376 11.8521 13.2424 11.6569L9.00048 7.414L4.75709 11.6569C4.56183 11.8521 4.24525 11.8521 4.04999 11.6569L3.69643 11.3033C3.50117 11.108 3.50117 10.7915 3.69643 10.5962L8.64618 5.64645C8.84144 5.45118 9.15803 5.45118 9.35329 5.64645L14.303 10.5962C14.4983 10.7915 14.4983 11.108 14.303 11.3033Z"
            fillRule="evenodd"
          />
          <g
            className="adui-icon-interactive-cover"
            fill="#000"
          >
            <path
              d="M14.303 11.3033L13.9495 11.6569C13.7542 11.8521 13.4376 11.8521 13.2424 11.6569L9.00048 7.414L4.75709 11.6569C4.56183 11.8521 4.24525 11.8521 4.04999 11.6569L3.69643 11.3033C3.50117 11.108 3.50117 10.7915 3.69643 10.5962L8.64618 5.64645C8.84144 5.45118 9.15803 5.45118 9.35329 5.64645L14.303 10.5962C14.4983 10.7915 14.4983 11.108 14.303 11.3033Z"
              fillRule="evenodd"
            />
          </g>
        </svg>
      </div>
      <div
        className="adui-numeric-input-button"
        onMouseDown={[Function]}
        role="none"
      >
        <svg
          className="adui-icon-base adui-icon-interactive adui-icon-interactive_level_2"
          data-icon="arrow-down"
          data-interactive={true}
          fill="var(--gray-700)"
          height={12}
          onClick={[Function]}
          viewBox="0 0 18 18"
          width={12}
        >
          <path
            d="M14.303 6.7L13.9495 6.34644C13.7542 6.15118 13.4376 6.15118 13.2424 6.34644L9.00048 10.5893L4.75709 6.34644C4.56183 6.15118 4.24525 6.15118 4.04999 6.34644L3.69643 6.7C3.50117 6.89526 3.50117 7.21184 3.69643 7.4071L8.64618 12.3569C8.84144 12.5521 9.15803 12.5521 9.35329 12.3569L14.303 7.4071C14.4983 7.21184 14.4983 6.89526 14.303 6.7Z"
            fillRule="evenodd"
          />
          <g
            className="adui-icon-interactive-cover"
            fill="#000"
          >
            <path
              d="M14.303 6.7L13.9495 6.34644C13.7542 6.15118 13.4376 6.15118 13.2424 6.34644L9.00048 10.5893L4.75709 6.34644C4.56183 6.15118 4.24525 6.15118 4.04999 6.34644L3.69643 6.7C3.50117 6.89526 3.50117 7.21184 3.69643 7.4071L8.64618 12.3569C8.84144 12.5521 9.15803 12.5521 9.35329 12.3569L14.303 7.4071C14.4983 7.21184 14.4983 6.89526 14.303 6.7Z"
              fillRule="evenodd"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
